ui/src/app/components/common-env-conf-panels/start-stop-environment-button.spec.tsx

```diff
@@ -0,0 +1,87 @@
+import '@testing-library/jest-dom';
+
+import { AppStatus, RuntimeStatus } from 'generated/fetch';
+
+import { render, screen, waitFor } from '@testing-library/react';
+import { UIAppType } from 'app/components/apps-panel/utils';
+
+import {
+  StartStopEnvironmentButton,
+  StartStopEnvironmentProps,
+} from './start-stop-environment-button';
+
+describe(StartStopEnvironmentButton.name, () => {
+  const runningText = 'Environment running, click to pause';
```

Comment:
I only tested the running and paused states here, because the others seemed trivial.